### ISSUE TITLE
SharePoint test plan intermittent fails MS.SHAREPOINT.4.2v1

### ISSUE THREAD

**Issue by crutchfield** [cisagov/ScubaGear#622]
# 🐛 Summary #

What's wrong? Please be specific.
SharePoint test plan intermittent fails MS.SHAREPOINT.4.2v1 due to assumption of sites being in an explicit order. 

## To reproduce ##

1. Run test plan against gcc high tenant

## Expected behavior ##

What did you expect to happen that didn't?

Test cases for MS.SHAREPOINT.4.2v1 to pass

## Any helpful log output or screenshots ##

Paste the results here:

```console

```

Add any screenshots of the problem here.

**Comment by crutchfield** [comment]
Consider filtering for correct site. 


`
 Get-SPOSite -Limit All | Where-Object {($_.Url -NotLike '*-my*') -And ($_.Url -Like '*/') }
`

**Comment by tkol2022** [comment]
### Reviewed 12/14
@crutchfield Can you please add a couple of sentences describing the problem in more detail?

Also I placed a priority of high because it looks like the Sharepoint test plan will not work on some tenants.

**Comment by crutchfield** [comment]
> ### Reviewed 12/14
> @crutchfield Can you please add a couple of sentences describing the problem in more detail?
> 
> Also I placed a priority of high because it looks like the Sharepoint test plan will not work on some tenants.

The Get-SPOSite returns a list of sites (including personal sites).  We are looking for a specific site and its associated attributes.  The query needs to filter for proper site and not just assume it is returned as teh first item in the array.

**Comment by mitchelbaker-cisa** [comment]
@schrolla We should consider adding this into Iceberg since it relates to SharePoint functional test plan improvements #1093.

**Comment by nanda-katikaneni** [comment]
Found the same issue while testing for Halibut release - policy 4.2 is consistently failing against GCCHIGH tenant. Screenshot below:
<img width="1121" alt="Screenshot 2024-06-05 at 5 25 15 PM" src="https://github.com/cisagov/ScubaGear/assets/104993331/99ec09d9-1400-43b9-bb7a-c9d5f3800ad7">

**Comment by tkol2022** [comment]
I reprioritized this because I was doing some testing with Sharepoint today on a different issue and I determined that the test plan for this policy affects the accuracy of the functional test in other tenants, not just GCC high. Therefore as the functional test is currently written, I don't have a high confidence that the functional test is working as it is supposed to even if you don't see any explicit errors in the command line when you run it. The fix is easy.
